src/Controllers/PiantaController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+  namespace App\Controllers;
+  use App\Data\Entities\Pianta;
+  use App\Data\Entities\StatoPianta;
+  use App\Data\Dao\PiantaDao as PiantaDao;
+  use App\Data\Dao\StatoPiantaDao as StatoPiantaDao;
+  use App\Controllers\QRCodeController as QRCodeController;
+  require_once __DIR__ . '/../Data/Entities/Pianta.php';
+  require_once __DIR__ . '/../Data/Dao/PiantaDao.php';
+  require_once __DIR__ . '/../Data/Entities/StatoPianta.php';
+  require_once __DIR__ . '/../Data/Dao/StatoPiantaDao.php';
+  require_once __DIR__ . '/QRCodeController.php';
+
+  class PiantaController{
+
+  //metodo per la get
+  public static function get($id)
+  {
+    $PiantaDao = new PiantaDao();
+    $StatoPiantaDao = new StatoPiantaDao();
+
+    $Pianta = $PiantaDao->getPianta($id);
+    //echo json_encode($Pianta);
+  }
+
+  //metodo per la PUT
+  public static function create()
+  {
+      
+    $post_json = json_decode(file_get_contents('php://input'));
+    $PiantaDao = new PiantaDao();
+    // plant creation
+    $Pianta = new Pianta($post_json->{"genere"}, $post_json->{"specie"}, $post_json->{"nome"}, $post_json->{"co2"}, $post_json->{"descrizione"});
+    $PiantaDao->store($Pianta);
+
+    $qrc = new QRCodeController();
+    $qrcode = $qrc->getQRCodeUrl($Pianta, 200, 200);
+    $Pianta->setQRCode($qrcode);
+    $PiantaDao->updateQRCode($Pianta);
```

Comment:
@shewraar in questo caso stai facendo due query sulla stessa tabella del database, prima inserendo le informazioni della pianta e poi facendo l'update del `QRCode` ... dovresti generare prima il QRcode, metterlo all'interno dell'oggetto e fare un'unica `store`.

Reply:
Si lo so ma se ci devo mettere l'id della pianta devo prima fare la query per l'inserimento della pianta e poi prendo l'id assegnato e lo inserisco nel qrcode così genero il link avente l'id della pianta appena creata.

Reply:
Va bene per il momento lasciala cosi allora